staging_service/import_specifications/individual_parsers.py

```diff
@@ -349,11 +351,33 @@ def parse_dts_manifest(path: Path) -> ParseResults:
     # dummy for now
     results = {}
     try:
+        jsonschema.Draft202012Validator.check_schema(dts_manifest_schema)
         with open(path, "r") as manifest:
             manifest_json = json.load(manifest)
         if not isinstance(manifest_json, dict):
-            errors.append(Error(ErrorType.PARSE_FAIL, "Manifest is not a dictionary", spcsrc))
-
+            return _error(Error(ErrorType.PARSE_FAIL, "Manifest is not a dictionary", spcsrc))
+        validator = jsonschema.Draft202012Validator(dts_manifest_schema)
+        for err in validator.iter_errors(manifest_json):
+            err_str = err.message
+            err_path = err.absolute_path
+            if err_path:
+                # paths can look like, say, ["instructions", "objects", 0, "data_type"]
+                # convert that '0' to "item 0" to be slightly more readable to users.
+                # kind of a mouthful below, but does that conversion in place
+                # The error above gets translated into:
+                # "<err.message> at instructions/objects/item 0/data_type"
+                #
+                # Another example would be if the path is just ["instructions"] and it's missing
+                # a property, the error might be:
+                # "missing property 'foo' for 'instructions'"
+                err_path = [f"item {elem}" if isinstance(elem, int) else elem for elem in err_path]
+                prep = "for"
+                if len(err_path) > 1:
+                    prep = "at"
+                err_str += f" {prep} {'/'.join(err_path)}"
```

Comment:
So... I wouldn't add this complication. People interacting with the API should know how to parse a json path and may even want to use code to parse it into a path segments, which won't work if there are non-standard elements in the path. Regular users aren't going to be submitting DTS manifests themselves, which means that they're not going to be able to fix errors - they'll need DTS or staging service code changes.

Rather than the for / at stuff, I'd probably just do `{err} at json path {path}`

Also, `/` is a valid character in a JSON string and so the `/` that are being added here are ambiguous. Not quite sure how to deal with that.

Maybe not now, but you might consider adding a path element to the returned error structure from the API.

Reply:
I'm mostly assuming the end user isn't going to know anything, they'll just get a manifest.json file, and load and use it. I'm also assuming they'll have minimal knowledge of the workings of a JSON file, so getting the jsonschema error dumped at them without context will be frustrating.

That said, this just parses that path list into a string. I can also just let Python stringify it and skip all this stuff, like you said.

Reply:
I would skip all this stuff. It's not helpful for the end user, adds complexity, and diverges from the standard representation of a json path

Reply:
Skipped.

Reply:
:+1: